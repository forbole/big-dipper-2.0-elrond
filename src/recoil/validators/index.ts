import {
  atomFamilyState,
} from './atom';

import {
  readValidator,
  writeValidator,
} from './selectors';

export {
  atomFamilyState,
  readValidator,
  writeValidator,
};
