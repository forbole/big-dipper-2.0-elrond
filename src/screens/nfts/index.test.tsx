import React from 'react';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';
import {
  MockTheme, wait,
} from '@tests/utils';
import Blocks from '.';

// ==================================
// unit tests
// ==================================
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
}));

jest.mock('./components', () => ({
  List: (props) => <div id="List" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Blocks', () => {
  it('matches snapshot', async () => {
    let component;
    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <MockTheme>
            <Blocks />
          </MockTheme>
        </RecoilRoot>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
