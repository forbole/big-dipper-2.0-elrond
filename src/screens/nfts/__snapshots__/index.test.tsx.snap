// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Blocks matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Layout"
  navTitle="nfts:nfts"
>
  <div
    id="List"
  />
</div>
`;
